File: spring-boot-socialnotes/src/main/java/com/socialnotes/service/ValutazioneService.java
package com.socialnotes.service;

import com.socialnotes.dao.ValutazioneDao;
import com.socialnotes.model.Valutazione;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class ValutazioneService {

    private final ValutazioneDao valutazioneDao;

    public ValutazioneService(ValutazioneDao valutazioneDao) { this.valutazioneDao = valutazioneDao; }

    public List<Valutazione> getValutazioniPost (String idPost) { return valutazioneDao.getValutazioniPost(idPost); }
    public boolean getValutazioneUtentePost (String nomeUtente, String idPost) { return valutazioneDao.getValutazioneUtentePost(nomeUtente, idPost); }
    public boolean setValutazione (Valutazione valutazione) { return valutazioneDao.setValutazione(valutazione); }
    public boolean deleteValutazione (String idValutazione) { return valutazioneDao.deleteValutazione(idValutazione); }
    public boolean deleteValutazioniPost (String idPost) { return valutazioneDao.deleteValutazioniPost(idPost); }

}

File: ClientApp/notes-app/src/app/service/valutazione.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import {Observable} from "rxjs";
import {Valutazione} from "../model/valutazione";

@Injectable({
  providedIn: 'root'
})
export class ValutazioneService {

  private URL = 'http://localhost:8080/valutazione';

  constructor(private http : HttpClient) { }

  //***************************************************
  //METODI HTTP VALUTAZIONE
  //TODO tutti i metodi sono da testare

  getValutazioniByPost(idPost : string): Observable<Valutazione>{
    return this.http.get<Valutazione>(this.URL + '/getValutazioniPost/' + idPost)
  }

  setValutazione(valutazione : Valutazione): Observable<Valutazione>{
    return this.http.post<Valutazione>(this.URL, valutazione)
  }

  deleteValutazione(id : string): Observable<unknown>{
    return this.http.delete(this.URL + '/deleteValutazione/' + id)
  }

  deleteValutazioniByPost(idPost : string): Observable<unknown>{
    return this.http.delete(this.URL + '/deleteValutazioniPost/' + idPost)
  }
}
